api/api_public_blockchain.go

```diff
@@ -504,9 +504,21 @@ type StructLogRes struct {
 }
 
 // formatLogs formats EVM returned structured logs for json output
-func FormatLogs(logs []vm.StructLog) []StructLogRes {
+func FormatLogs(timeout time.Duration, logs []vm.StructLog) ([]StructLogRes, error) {
+	logTimeout := false
+	deadlineCtx, cancel := context.WithTimeout(context.Background(), timeout)
+	go func() {
+		<-deadlineCtx.Done()
+		logger.Error("[Trace logger] Timeout reached", "timeout", timeout, "err", deadlineCtx.Err())
+		logTimeout = true
+	}()
+	defer cancel()
+
 	formatted := make([]StructLogRes, len(logs))
 	for index, trace := range logs {
+		if logTimeout {
+			return nil, fmt.Errorf("[Trace logger] Log truncated by timeout")
```

Comment:
```suggestion
			return nil, fmt.Errorf("trace logger timeout")
```
Can we match the error style with others? (e.g. `execution timeout`)

Reply:
Thanks. I missed fixing the wrong msg. We may consider a new logging format that abstracts where its logging is involved by adding a prefix.